includes/Products.php

```diff
@@ -193,16 +186,20 @@ public static function disable_sync_for_products_with_terms( array $args ) {
 	/**
 	 * Determines whether the given product should be synced.
 	 *
-	 * @see Products::published_product_should_be_synced()
+	 * @deprecated use \SkyVerge\WooCommerce\Facebook\ProductSync\ProductValidator::validate() instead
 	 *
 	 * @since 1.10.0
 	 *
 	 * @param \WC_Product $product
 	 * @return bool
 	 */
 	public static function product_should_be_synced( \WC_Product $product ) {
-
-		return 'publish' === $product->get_status() && self::published_product_should_be_synced( $product );
+		try {
+			facebook_for_woocommerce()->get_product_sync_validator( $product )->validate();
+			return true;
+		} catch ( \Exception $e ) {
+			return false;
```

Comment:
I guess could log here if verbose debug logging is enabled - make use of those exception messages. This would make things a bit more transparent too, so merchants can diagnose why some product isn't syncing.

Reply:
Please don't do that. You don't want to log here. There are legitimate reasons why products should not be synced that do not issue. Global sync disabled, product sync disabled, excluded categories. These may be intentional settings. 

If we want to use it we should be a bit more direct about this. I will create a PR that will use this information for the

Reply:
Please check https://github.com/woocommerce/facebook-for-woocommerce/pull/1937. This is how we should utilize the information from this code.

Reply:
Adding logging here would be very noisy due to the frequency of daily feed generation so we might not want to do that. Although it could be useful for debugging/dev purposes. Something like #1937 is a nicer solution for users.